node/core/pvf/src/execute/queue.rs

```diff
@@ -322,6 +413,16 @@ fn assign(queue: &mut Queue, worker: Worker, job: ExecuteJob) {
 		"assigning the execute worker",
 	);
 
+	debug_assert_eq!(
+		queue
+			.workers
+			.running
+			.get(worker)
+			.expect("caller must provide existing worker; qed")
+			.executor_params_hash,
+		job.executor_params.hash()
+	);
```

Comment:
Please document that this function requires a Job whose params match the worker. Right now the doc just says `The worker must be running and idle.` and doesn't mention the job.

As an aside, I've always felt that the whole PVF code is quite brittle, with lots of assumptions baked-in everywhere. It makes it hard to reason about.

Reply:
It's an excellent idea to use types wherever possible; I've added it to #6474. But I believe it's not possible for worker-job relationships (or maybe I just cannot imagine how it should work, we don't have a complete set of possible job types at compile time). So we can continue on that in #6474.